File: java/WebScraper.java
/**
 * JSoup-powered helpers for fetching pages and extracting elements with robust error handling.
 */

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class WebScraper {
    public static void main(String[] args) throws Exception {
        String url = "https://example.com";
        // Connect to the website and parse HTML
        Document doc = Jsoup.connect(url).get();
        // Select all h1 elements
        Elements headings = doc.select("h1");
        // Print each heading text
        for (Element h1 : headings) {
            System.out.println(h1.text());
        }
    }
}

File: python/web_scraper.py
"""Lightweight HTML scraping helpers using requests and BeautifulSoup.

Provides a small toolkit for fetching pages, parsing elements, and extracting
links or text content for quick data collection scripts.
"""

import requests
from bs4 import BeautifulSoup


def fetch_page(url):
    response = requests.get(url)
    response.raise_for_status()
    return response.text


def parse_content(html):
    soup = BeautifulSoup(html, 'html.parser')
    titles = [h.get_text(strip=True) for h in soup.find_all('h1')]
    return titles


if __name__ == "__main__":
    url = "https://example.com"
    html = fetch_page(url)
    data = parse_content(html)
    for item in data:
        print(item)
